felix/Makefile

```diff
@@ -133,10 +133,10 @@ sub-build-%:
 bin/calico-felix: bin/calico-felix-$(ARCH)
 	ln -f bin/calico-felix-$(ARCH) bin/calico-felix
 
-libbpf: $(LIBBPF_PATH)/libbpf.a
-$(LIBBPF_PATH)/libbpf.a: ../go.mod
-	$(MAKE) mod-download
+libbpf:
 	$(DOCKER_GO_BUILD) sh -c "make -j 16 -C bpf-gpl/include/libbpf/src BUILD_STATIC_ONLY=1"
+$(LIBBPF_PATH)/libbpf.a: libbpf ../go.mod
```

Comment:
This depends on a phony target so it will _always_ get rebuilt.  I think the right thing to do would be to depend on all the .c/.h files in the libbpf directory.  Should just be a matter of doing a wildcard.

Reply:
Yes, that is an option, but since the lib has its own Makefile which does exactly that, it seems like the right and nicer thing to execute that rather than doing everything in the top Makefile. the `libbpf` target is only do descent.

Reply:
The problem is that, this way, `$(LIBBPF_PATH)/libbpf.a` will always be stale so targets that depend on it will always get rebuilt.  It's an annoying problem with `make` that I don't have a great solution for

Reply:
yeah :-/